src/PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.hpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"     // IWYU pragma: keep
+#include "Elliptic/Systems/Poisson/Tags.hpp"    // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Poisson {
+/*!
+ * \brief Analytic solutions to the Poisson equation.
+ */
+namespace Solutions {
+/*!
+ * \brief A solution to the Poisson equation with a discontinuous first
+ * derivative.
```

Comment:
Add analytic solution + source in documentation
Is there a reference for this solution? where does it come from/does it appear anywhere else

Reply:
Added this, but left out the source since it's a very long and messy polynomial, in particular in 2 dimensions (you can see it in the python test). It's just the laplacian of the field, so easily calculated by copying the field into a symbolic calculator.